src/Access/AccessRights.cpp

```diff
@@ -1457,18 +1463,18 @@ bool AccessRights::isGrantedImpl(const AccessFlags & flags, const Args &... args
 template <bool grant_option>
 bool AccessRights::containsImpl(const AccessRights & other) const
 {
-    auto helper = [&](const std::unique_ptr<Node> & root_node) -> bool
+    auto helper = [&](const std::unique_ptr<Node> & root_node, const std::unique_ptr<Node> & other_root_node) -> bool
     {
         if (!root_node)
             return !other.root;
-        if (!other.root)
+        if (!other_root_node)
             return true;
-        return root_node->contains(*other.root);
+        return root_node->contains(*other_root_node);
     };
     if constexpr (grant_option)
-        return helper(root_with_grant_option);
+        return helper(root_with_grant_option, other.root);
     else
-        return helper(root);
+        return helper(root, other.root) && helper(root_with_grant_option, other.root_with_grant_option);
```

Comment:
That look wrong, shouldn't it be
```
    if constexpr (grant_option)
        return helper(root_with_grant_option, other.root_with_grant_option);
    else
        return helper(root, other.root);
```
?

Reply:
Ok, I see it now, but let's write some comments about this behavior before the declaration of `AccessRights::contains()` and `AccessRights::containsWithGrantOption()`.